plugins/woocommerce/client/blocks/assets/js/blocks/add-to-cart-with-options/grouped-product-selector/product-item/edit.tsx

```diff
@@ -152,6 +156,10 @@ export default function ProductItemTemplateEdit(
 	const [ selectedProductItem, setSelectedProductItem ] =
 		useState< number >();
 
+	if ( ! products ) {
```

Comment:
NIT: If the store doesn't have products, the spinner will always be present.

Reply:
Yes, I agree. Not displaying the blocks at all or displaying the spinner feels wrong in both cases.

I opened an issue about this here: https://github.com/woocommerce/woocommerce/issues/61252. Ideally, we should display sample product data.